File: app/src/main/java/net/squanchy/settings/SettingsInjector.java
package net.squanchy.settings;

import android.app.Activity;
import android.support.v7.app.AppCompatActivity;

import net.squanchy.injection.ActivityContextModule;
import net.squanchy.injection.ApplicationInjector;
import net.squanchy.navigation.NavigationModule;
import net.squanchy.signin.SignInModule;

final class SettingsInjector {

    private SettingsInjector() {
        // no instances
    }

    static SettingsFragmentComponent obtainForFragment(AppCompatActivity activity) {
        return DaggerSettingsFragmentComponent.builder()
                .activityContextModule(new ActivityContextModule(activity))
                .applicationComponent(ApplicationInjector.obtain(activity))
                .navigationModule(new NavigationModule())
                .signInModule(new SignInModule())
                .build();
    }

    static SettingsActivityComponent obtainForActivity(Activity activity) {
        return DaggerSettingsActivityComponent.builder()
                .applicationComponent(ApplicationInjector.obtain(activity))
                .signInModule(new SignInModule())
                .build();
    }
}

File: app/src/main/java/net/squanchy/settings/SettingsInjector.kt
package net.squanchy.settings

import android.app.Activity
import android.support.v7.app.AppCompatActivity

import net.squanchy.injection.ActivityContextModule
import net.squanchy.injection.applicationComponent
import net.squanchy.navigation.NavigationModule
import net.squanchy.signin.SignInModule

internal fun settingsFragmentComponent(activity: AppCompatActivity): SettingsFragmentComponent {
    return DaggerSettingsFragmentComponent.builder()
        .activityContextModule(ActivityContextModule(activity))
        .applicationComponent(activity.applicationComponent)
        .navigationModule(NavigationModule())
        .signInModule(SignInModule())
        .build()
}

internal fun settingsActivityComponent(activity: Activity): SettingsActivityComponent {
    return DaggerSettingsActivityComponent.builder()
        .applicationComponent(activity.applicationComponent)
        .signInModule(SignInModule())
        .build()
}